src/Components/GEOSldas_GridComp/GEOSlandassim_GridComp/clsm_ensupd_upd_routines.F90

```diff
@@ -4403,6 +4406,194 @@ subroutine cat_enkf_increments(                               &
        end do
 
        ! ----------------------------------
+
+    case (13) select_update_type   ! 3d soil moisture/Tskin/ght(1) analysis; Tb+sfmc+sfds obs
+       
+       ! combines update_types 2 and 10
+       !
+       ! update each tile separately using all observations within customized halo around each tile
+       !
+       ! amfox, 14 September 2023
+       
+       if (logit) write (logunit,*) 'get 3d soil moisture/Tskin/ght(1) increments; Tb+sfmc obs'
+       
+       N_select_varnames  = 0
+       
+       if (any(obs_param%varname == 'Tb')) then
```

Comment:
Do we need the "trim()" function here, and  similarly in the next two blocks?  I wonder if the "==" statement reliably produces the desired result across compilers, given that we should have trailing whitespace in %varname.
Note that obs_param is dimension(N_obs_param).  Not sure if trim() works on such a 1d array.